File: pharmacy-api/local-api/broker_setup.go
package main

import (
	"log"

	models "pharmacy-api/shared/models"
)

func setupBroker(broker *models.Broker) (err error) {

	err = broker.Channel.ExchangeDeclare("local_exchange", "direct", true, false, false, false, nil)
	if err != nil {
		log.Fatal("failed to declare exchange:", err)
		return
	}

	queues := []struct {
		name       string
		routingKey string
	}{
		{"local_goods_queue", "goods"},
		{"local_schedule_queue", "schedule"},
		{"local_orders_queue", "orders"},
		{"local_announces_queue", "announces"},
		{"local_users_queue", "users"},
		{"local_roles_queue", "roles"},
	}

	for _, q := range queues {
		queue, err := broker.Channel.QueueDeclare(q.name, true, false, false, false, nil)
		if err != nil {
			log.Fatal("failed at declare queues:", err)
			return err
		}

		err = broker.Channel.QueueBind(queue.Name, q.routingKey, "local_exchange", false, nil)
		if err != nil {
			log.Fatal("failed at bindings queues:", err)
			return err
		}
	}

	log.Println("RabbitMQ setup completed successfully")
	return
}

File: pharmacy-api/src/local-api/broker_setup.ts
import { Channel } from 'amqplib';
import { Broker } from '@pharmacy/src/shared/models/models';

export async function setupBroker(broker: Broker): Promise<void> {
  try {
    await broker.channel.assertExchange("local_exchange", "direct", {
      durable: true,
      autoDelete: false,
    });

    const queues = [
      { name: "local_goods_queue", routingKey: "goods" },
      { name: "local_schedule_queue", routingKey: "schedule" },
      { name: "local_orders_queue", routingKey: "orders" },
      { name: "local_announces_queue", routingKey: "announces" },
      { name: "local_users_queue", routingKey: "users" },
      { name: "local_roles_queue", routingKey: "roles" },
    ];

    for (const q of queues) {
      const queue = await broker.channel.assertQueue(q.name, {
        durable: true,
        autoDelete: false,
      });

      await broker.channel.bindQueue(queue.queue, "local_exchange", q.routingKey);
    }

    console.log("RabbitMQ setup completed successfully");
    console.log(`Exchange "local_exchange" declared with ${queues.length} queues bound`);
  } catch (err) {
    console.error("Failed to setup RabbitMQ:", err);
    throw err;
  }
}